ttml.go

```diff
@@ -70,35 +70,48 @@ type TTMLInMetadata struct {
-	FontSize        string `xml:"fontSize,attr,omitempty"`
-	FontStyle       string `xml:"fontStyle,attr,omitempty"`
-	FontWeight      string `xml:"fontWeight,attr,omitempty"`
-	LineHeight      string `xml:"lineHeight,attr,omitempty"`
-	Opacity         string `xml:"opacity,attr,omitempty"`
-	Origin          string `xml:"origin,attr,omitempty"`
-	Overflow        string `xml:"overflow,attr,omitempty"`
-	Padding         string `xml:"padding,attr,omitempty"`
-	ShowBackground  string `xml:"showBackground,attr,omitempty"`
-	TextAlign       string `xml:"textAlign,attr,omitempty"`
-	TextDecoration  string `xml:"textDecoration,attr,omitempty"`
-	TextOutline     string `xml:"textOutline,attr,omitempty"`
-	UnicodeBidi     string `xml:"unicodeBidi,attr,omitempty"`
-	Visibility      string `xml:"visibility,attr,omitempty"`
-	WrapOption      string `xml:"wrapOption,attr,omitempty"`
-	WritingMode     string `xml:"writingMode,attr,omitempty"`
-	ZIndex          int    `xml:"zIndex,attr,omitempty"`
+	BackgroundColor *string `xml:"backgroundColor,attr,omitempty"`
+	Color           *string `xml:"color,attr,omitempty"`
+	Direction       *string `xml:"direction,attr,omitempty"`
+	Display         *string `xml:"display,attr,omitempty"`
+	DisplayAlign    *string `xml:"displayAlign,attr,omitempty"`
+	Extent          *string `xml:"extent,attr,omitempty"`
+	FontFamily      *string `xml:"fontFamily,attr,omitempty"`
+	FontSize        *string `xml:"fontSize,attr,omitempty"`
+	FontStyle       *string `xml:"fontStyle,attr,omitempty"`
+	FontWeight      *string `xml:"fontWeight,attr,omitempty"`
+	LineHeight      *string `xml:"lineHeight,attr,omitempty"`
+	Opacity         *string `xml:"opacity,attr,omitempty"`
+	Origin          *string `xml:"origin,attr,omitempty"`
+	Overflow        *string `xml:"overflow,attr,omitempty"`
+	Padding         *string `xml:"padding,attr,omitempty"`
+	ShowBackground  *string `xml:"showBackground,attr,omitempty"`
+	TextAlign       *string `xml:"textAlign,attr,omitempty"`
+	TextDecoration  *string `xml:"textDecoration,attr,omitempty"`
+	TextOutline     *string `xml:"textOutline,attr,omitempty"`
+	UnicodeBidi     *string `xml:"unicodeBidi,attr,omitempty"`
+	Visibility      *string `xml:"visibility,attr,omitempty"`
+	WrapOption      *string `xml:"wrapOption,attr,omitempty"`
+	WritingMode     *string `xml:"writingMode,attr,omitempty"`
+	ZIndex          *int    `xml:"zIndex,attr,omitempty"`
 }
 
 // StyleAttributes converts TTMLInStyleAttributes into a StyleAttributes
 func (s TTMLInStyleAttributes) styleAttributes() (o *StyleAttributes) {
-	o = &StyleAttributes{
+	o = getStyleAttributes(s)
+	o.propagateTTMLAttributes()
+	return
+}
+
+// StyleAttributes converts TTMLInStyleAttributes for region into a StyleAttributes
+func (s TTMLInStyleAttributes) regionStyleAttributes() (o *StyleAttributes) {
+	o = getStyleAttributes(s)
+	o.propagateTTMLRegionAttributes()
+	return
+}
+
+func getStyleAttributes(s TTMLInStyleAttributes) *StyleAttributes {
```

Comment:
I'd rather rename this function to something more TTML-specific and more self-explanatory like `newStyleAttributesFromTTMLInStyleAttributes`

Reply:
done